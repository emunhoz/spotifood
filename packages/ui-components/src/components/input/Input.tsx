import React, { ChangeEvent, InputHTMLAttributes } from 'react'
import { Label } from '../../'
import * as S from './Input.style'

export interface InputTextProps extends InputHTMLAttributes<HTMLInputElement> {
  disabled?: boolean
  /**
   * Input value
   */
  value?: string
  /**
   * Placeholder for input
   */
  placeholder?: string
  /**
   * Label for input
   */
  label?: string
  /**
   * onChange callback
   */
  onChange: (event: ChangeEvent<HTMLInputElement>) => void
  /**
   * State style when there is an error
   */
  error?: boolean
  /**
   * Message to help user
   */
  message?: string
}

export const Input = ({
  value,
  placeholder,
  disabled,
  label,
  error,
  message = '',
  ...rest
}: InputTextProps) => (
  <S.InputWrapper>
    <Label value={label} htmlFor={label} />
    <S.Input
      id={label}
      name={label}
      value={value}
      disabled={disabled}
      placeholder={placeholder}
      error={!!error}
      {...rest}
    />
    <S.HelpMessage error={!!error}>{message}</S.HelpMessage>
  </S.InputWrapper>
)
