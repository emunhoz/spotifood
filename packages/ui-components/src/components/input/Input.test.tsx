import React from 'react'
import { render } from '../../utils/test-utils'
import { Input } from './Input'

describe('[Component]: Input', () => {
  test('render component', () => {
    const { asFragment } = render(<Input onChange={console.log} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('render component with error style', () => {
    const { asFragment } = render(<Input error message="Help message" onChange={console.log} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
