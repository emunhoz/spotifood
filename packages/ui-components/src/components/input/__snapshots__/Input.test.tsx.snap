// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Input render component 1`] = `
<DocumentFragment>
  <div
    class="sc-ezrdKe hTnOjt"
  >
    <label
      class="sc-hBEYos iAuSGj"
    />
    <input
      class="sc-bYEvPH jcbMNy"
      value=""
    />
    <span
      class="sc-kLgntA iKbHJN"
    />
  </div>
</DocumentFragment>
`;

exports[`[Component]: Input render component with error style 1`] = `
<DocumentFragment>
  <div
    class="sc-ezrdKe hTnOjt"
  >
    <label
      class="sc-hBEYos iAuSGj"
    />
    <input
      class="sc-bYEvPH aOmjA"
      value=""
    />
    <span
      class="sc-kLgntA eyXmyp"
    >
      Help message
    </span>
  </div>
</DocumentFragment>
`;
